plugins/jobs/src/js/components/JobFormModal.tsx

```diff
@@ -102,39 +145,251 @@ export default class JobFormModal extends React.Component<
     isEdit: false,
     job: new Job()
   };
+  private triggerFormSubmit: FormSubmitCallback;
 
   constructor() {
-    super(...arguments);
+    super();
 
     this.state = {
-      defaultTab: ""
+      jobFormModel: null,
+      jobJsonString: null,
+      isJsonMode: false
     };
 
-    this.handleTabChange = this.handleTabChange.bind(this);
+    this.triggerFormSubmit = undefined;
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleJSONEditorChange = this.handleJSONEditorChange.bind(this);
+    this.handleFormChange = this.handleFormChange.bind(this);
+    this.handleTriggerSubmit = this.handleTriggerSubmit.bind(this);
+    this.handleInputModeToggle = this.handleInputModeToggle.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps: JobFormModalProps) {
+    if (nextProps.isOpen !== this.props.isOpen) {
+      this.resetState();
```

Comment:
🎨 not sure if this is legacy code, but can we rename this method (`resetState`) to `updateJobFromProps` or similar?

Reply:
or simply move the code here, looks like this the only place where it is called…